### DISCUSSION TITLE
[archived] "Cron Workflows" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#11428] [General]
# argo-workflows/cron-workflows/

https://argoproj.github.io/argo-workflows/cron-workflows/

**Comment by atharvatessell** [comment]
From the code given above, I am getting error
`Bad Request: spec.entrypoint is required`

**Reply by atharvatessell** [reply]
Solved
I was trying to run it in `Workflows` tab in UI

**Comment by sstaley-hioscar** [comment]
Hi,

Does Argo Workflows utilize the same Kubernetes scheduler for crons as a Kubernetes cron job? I understand Kubernetes runs into some limitations at scale with respect to crons and want to understand if Argo Workflows would encounter the same problem.

**Reply by terrytangyuan** [reply]
This is used to schedule workflows https://github.com/robfig/cron

**Reply by sstaley-hioscar** [reply]
Thank you. Where is the cron state saved? Etcd?

**Reply by terrytangyuan** [reply]
Yes, the status of a CronWorkflow is saved in its status field. CronWorkflow is a custom resource so it will be persisted in etcd.

**Reply by sstaley-hioscar** [reply]
Thanks again @terrytangyuan. Any chance you'd have any experience running crons at scale with Workflows? Would you know if it handles better than the K8s controller? I'm speaking of the issue discussed here:
* https://eng.lyft.com/improving-kubernetes-cronjobs-at-scale-part-1-cf1479df98d4
* https://eng.lyft.com/how-we-learned-to-improve-kubernetes-cronjobs-at-scale-part-2-of-2-dad0c973ffca

**Reply by terrytangyuan** [reply]
Thanks for sharing. It would be a good read for the community to see if there would be room for optimizations of the current CronController.
